doc/Developing.md

```diff
@@ -236,22 +236,9 @@ merge them into the release branch.
 Following the release, you need to merge `RELEASE_NOTES.md` from the release branch into develop. If you released from 
 `HEAD`, then you also need to bump the SNAPSHOT versions in the `develop` branch using the `scripts/update_version.sh` script.
 
-### Releasing manually
+### Creating draft release for testing
 
-This should only be done if you want to create a debug release for internal testing (and even then, you can follow the automated process above and just not publish the draft release).
```

Comment:
Simplified the instructions, since using the automated (GHA workflow) is recommended anyway.